src/ros_filter.cpp

```diff
@@ -2318,6 +2318,11 @@ namespace RobotLocalization
         else
         {
           tf2::fromMsg(msg->orientation, curAttitude);
+          if (fabs(curAttitude.length() - 1.0) > 0.01)
```

Comment:
Do we know what kind of tolerance for normalization we require? The PR job passed, so this isn't affecting any of the bags in the test data set, which means that either their quaternion lengths were within this tolerance, or the normalization doesn't have a strong effect (I'm assuming the former).